addons/mail/static/src/new/messaging_menu/messaging_menu.js

```diff
@@ -85,11 +109,23 @@ export class MessagingMenu extends Component {
             );
         });
         const tab = this.store.discuss.activeTab;
-        if (tab === "all") {
-            return previews;
-        }
-        const target = this.tabToThreadType(tab);
-        return previews.filter((preview) => target.includes(preview.type));
+        const previews =
+            tab === "all"
+                ? threads
+                : threads.filter(({ type }) => this.tabToThreadType(tab).includes(type));
+        return previews.map((thread) => {
+            const message = thread.mostRecentNeedactionMsg ?? thread.mostRecentMsg;
```

Comment:
This does not look correct, for channels the message displayed is always the `mostRecentMsg`, and if there is also a `mostRecentNeedactionMsg` it should be a separate item.